pkg/backend_test.go

```diff
@@ -0,0 +1,115 @@
+package pkg
```

Comment:
this is a great unit test instead of relying on command line test.  please move this to cmd/compile_test.go